test/TPPCore.Service.Chat.Tests/DummyRestfulTest.cs

```diff
@@ -43,70 +39,45 @@ private ServiceTestRunner newServiceRunner()
         [Fact]
         public async Task TestUserIdUsername()
         {
-            var runner = newServiceRunner();
+            var runner = NewServiceRunner();
             await runner.SetUpAsync(getOptions());
 
             var httpClient = runner.Runner.Context.RestfulClient;
-            var uri = new Uri(
-                runner.Runner.Context.RestfulServer.Context.GetUri(),
-                "client/dummyTest/username");
-            var result = await httpClient.GetJsonAsync(uri);
-            Assert.Equal(HttpStatusCode.OK, result.Response.StatusCode);
-
-            var jsonDoc = result.JsonDoc;
-            var username = jsonDoc.Value<string>("username");
-            Assert.Equal("dummy", username);
-
-            uri = new Uri(
-                runner.Runner.Context.RestfulServer.Context.GetUri(),
-                "client/dummyTest/user_id");
-            result = await httpClient.GetJsonAsync(uri);
-            Assert.Equal(HttpStatusCode.OK, result.Response.StatusCode);
-
-            jsonDoc = result.JsonDoc;
-            var userId = jsonDoc.Value<string>("userId");
-            Assert.Equal("dummy", userId);
+            ChatClient client = new ChatClient(runner.Runner.Context.RestfulServer.Context.GetUri().ToString(), "dummyTest", httpClient);
+            var result = await client.GetUserName();
+            Assert.Equal("dummy", result);
+
+            result = await client.GetUserId();
+            Assert.Equal("dummy", result);
 
             await runner.TearDownAsync();
         }
 
         [Fact]
         public async Task TestRoomList()
         {
-            var runner = newServiceRunner();
+            var runner = NewServiceRunner();
             await runner.SetUpAsync(getOptions());
 
             var httpClient = runner.Runner.Context.RestfulClient;
-            var uri = new Uri(
-                runner.Runner.Context.RestfulServer.Context.GetUri(),
-                "chat/dummyTest/%23somechannel/room_list");
-            var result = await httpClient.GetJsonAsync(uri);
-
-            Assert.Equal(HttpStatusCode.OK, result.Response.StatusCode);
-
-            var jsonDoc = result.JsonDoc;
-            var users = jsonDoc.Value<JArray>("users");
-
-            Assert.Equal(2, users.Count);
-            Assert.Equal("dummy", (string) users.SelectToken("[0].username"));
+            ChatClient client = new ChatClient(runner.Runner.Context.RestfulServer.Context.GetUri().ToString(), "dummyTest", httpClient);
+            var result = await client.GetRoomList("%23somechannel");

```

Comment:
It should be `#somechannel` as in IRC so the user doesn't have to know they need to percent encode the parameters.